packages/html/src/HTMLTooltip.ts

```diff
@@ -45,11 +55,35 @@ export class HTMLTooltip extends HTMLWidget {
 
     update(domNode, element) {
         super.update(domNode, element);
+
+        if(this._contentNode !== this._prevContentNode){
+            const node = this._tooltipElement.node();
+            [...node.querySelectorAll("*")]
+            .map(n=>n.__data__)
+            .filter(n=>n)
+            .forEach(w=>{
+                if(typeof w.target === "function"){
+                    w.target(null);
+                }
+                if(typeof w.exit === "function"){
+                    w.exit();
+                }
+            });
+            node.innerHTML = "";
+            node.appendChild(this._contentNode);
+            this._prevContentNode = this._contentNode;
+        }
+
+        if (this._contentNode) {
+            this.onShowContent(this._contentNode);
```

Comment:
What is this supposed to do?

Reply:
It's a hook for shown content (aka a callback that runs when the tooltip is displayed via _contentNode rather than html). I added a gallery sample where I use it to render a Pie chart in the tooltip:
demos/gallery/samples/misc/React Timeline Tooltip Content.js